android/java/org/chromium/chrome/browser/ntp/BraveNewTabPageLayout.java

```diff
@@ -909,9 +908,12 @@ private void runFeed(boolean isNewContent, Feed feed) {
         }
 
         mFeedHash = feed.hash;
-        mNtpAdapter.notifyItemRangeRemoved(
-                mNtpAdapter.getStatsCount() + mNtpAdapter.getTopSitesCount() + 1,
-                mNewsItemsFeedCard.size());
+        new Handler(Looper.getMainLooper()).post(() -> {
```

Comment:
I don't think the logic is pretty good as it looks like the code after that line suppose that the adapter is already updated or I get it wrong? Perhaps a good idea is to revert the original PR to prevent crashing and do refactor of that place to separate UI logic and the rest that could be done in the background thread.

Reply:
updated